packages/web/components/funnels/concentrated-liquidity/supercharge-dai-osmo-pool.tsx

```diff
@@ -0,0 +1,84 @@
+import classNames from "classnames";
+import Image from "next/image";
+import { FunctionComponent } from "react";
+
+import { Button } from "~/components/buttons";
+import { CustomClasses } from "~/components/types";
+import { useWindowSize } from "~/hooks";
+
+export const SuperchargeDaiOsmoPool: FunctionComponent<
+  {
+    title: string;
+    caption?: string;
+    primaryCta: string;
+    secondaryCta: string;
+    onCtaClick: () => void;
+    onSecondaryClick: () => void;
+  } & CustomClasses
+> = ({
+  className,
+  title,
+  caption,
+  primaryCta,
+  secondaryCta,
+  onCtaClick,
+  onSecondaryClick,
+}) => {
+  const { isMobile } = useWindowSize();
+
+  const Buttons = () => (
```

Comment:
The reason I placed them here is just semantics, but since they share so many of the parent props and variables, I just decided to inline it in the body and still avoid repeating the markup below where a different structure is used for mobile.

If you still disagree, I'm down to extract it in the same file and pass it it's own props.

Reply:
It's definitely more performant to define them outside of the component. In general, there are cases where it's not necessary to optimize for performance and it's worth the hit if it helps devEx, but I'm not sure that's a case here ( I'm not sure how often this rerenders)? I think it makes sense to define outside the component so we don't have to redefine on every render.